src/libraries/Microsoft.Extensions.Logging.Console/tests/Microsoft.Extensions.Logging.Console.Tests/ConsoleFormatterTests.cs

```diff
@@ -27,7 +27,43 @@ protected string GetMessage(List<ConsoleContext> contexts)
             return string.Join("", contexts.Select(c => c.Message));
         }
 
-        internal static (ConsoleLogger Logger, ConsoleSink Sink, ConsoleSink ErrorSink, Func<LogLevel, string> GetLevelPrefix, int WritesPerMsg) SetUp(
+        internal class SetupDisposeHelper : IDisposable
+        {
+            public ConsoleLogger Logger;
+            public ConsoleSink Sink;
+            public ConsoleSink ErrorSink;
+            public Func<LogLevel, string> GetLevelPrefix;
+            public int WritesPerMsg;
+            public TestLoggerProcessor LoggerProcessor;
+            public SetupDisposeHelper(ConsoleLogger logger, ConsoleSink sink, ConsoleSink errorSink, Func<LogLevel, string> getLevelPrefix, int writesPerMsg, TestLoggerProcessor loggerProcessor)
+            {
+                Logger = logger;
+                Sink = sink;
+                ErrorSink = errorSink;
+                GetLevelPrefix = getLevelPrefix;
+                WritesPerMsg = writesPerMsg;
+                LoggerProcessor = loggerProcessor;
+            }
+
+            private bool _isDisposed;
+
+            protected virtual void Dispose(bool disposing)
+            {
+                if (!_isDisposed)
+                {
+                    LoggerProcessor.Dispose();
+                    _isDisposed = true;
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(disposing: true);
+                GC.SuppressFinalize(this);
+            }
+        }
```

Comment:
This can be simplified. It doesn't have a finalizer, so you don't need to implement the dispose pattern, and it can be a record to avoid needing the separate ctor, e.g.
```C#
internal sealed record class SetupDisposeHelper(
    ConsoleLogger Logger,
    ConsoleSink Sink,
    Func<LogLevel, string> GetLevelPrefix,
    int WritesPerMsg,
    TestLoggerProcessor LoggerProcessor) : IDisposable
{
    public void Dispose() => LoggerProcessor.Dispose();
}
```
Same for the one in the other file.

Reply:
record complained something about missing attribute for net4 in my VS

Reply:
If they are the same in `ConsoleLoggerTest` and `ConsoleFormatterTests`, what about creating `ConsoleTestsBase` and inherit from it?